src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceTest.java

```diff
@@ -0,0 +1,93 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.scm.SCM;
+import hudson.util.LogTaskListener;
+import jenkins.scm.api.SCMHeadObserver;
+import jenkins.scm.api.SCMHeadObserver.Collector;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMRevision;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.contains;
+
+public class GitHubSCMSourceTest {
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(8089);
+
+
+    @Rule
+    public JenkinsRule jrule = new JenkinsRule();
+
+
+    @Test
+    public void testRetrieve() throws Exception {
+        GitHubSCMSource ghss = new GitHubSCMSource("testSource", "http://localhost:8089/", null, null, "testOwner", "testRepo");
```

Comment:
Perhaps we need to create an actual access token, test everything with real requests and responses, then invalidate the token before pushing.